### PR TITLE
Make it buildable with refsw 18.3 and Playready 4

### PR DESCRIPTION
**Pull request by dwrobel**
- Drop WPEFramework dependency,
- Drop NEXUS_PLAYREADY_SVP_ENABLE option switch,
- Rely on Playready pkg-config provided by the platform
  -- It is now a platform which defines whether SAGE is
     ON or OFF by exporting:
         -DCMD_DRM_PLAYREADY_SAGE_IMPL
- Fix compilation issues.

### CODE REVIEW

File: MediaSession.h [hunk 1]
```diff
@@ -16,21 +16,30 @@
...
-#include <refsw/bkni.h>
-
-#include <refsw/oemcommon.h>
-#include <refsw/drmmanager.h>
-#include <refsw/drmmathsafe.h>
-#include <refsw/drmtypes.h>
-#include <refsw/drmerr.h>
+// Comes from Nexus configuration
+// passed by playready pkg-config file.
+#ifdef CMD_DRM_PLAYREADY_SAGE_IMPL
```
Comment (pwielders) [inline MediaSession.h]: Lets cleanup the code a bitmore and use the CMD_DRM_PLAYREADY_SAGE_IMPL in stead of the NEXUS_PLAYREADY_SVP_ENABLE flag.

Comment (dwrobel) [reply MediaSession.h]: Done

**Comment by bramoosterhuis** [comment]
@dwrobel  Why did you reintroduce the old cmake structure. It breaks my vodafone build?

**Comment by dwrobel** [comment]
> @dwrobel Why did you reintroduce the old cmake structure. It breaks my vodafone build.

What it means old structue?

As per git message: "Rely on Playready pkg-config provided by the platform". So, is your platform providing a `playready4` package-config?

**Comment by bramoosterhuis** [comment]
> > @dwrobel Why did you reintroduce the old cmake structure. It breaks my vodafone build.
> 
> What it means old structue?
> 
> As per git message: "Rely on Playready pkg-config provided by the platform". So, is your platform providing a `playready4` package-config?

No it's not providing playready 4. now it cannot find nexus headers. Check the the changes in the CMakeList.txt.... is completely back to before my refactor... was that really needed to only add playready 4 support...

**Comment by dwrobel** [comment]
I assume once your `refsw` will provide the package-config you shouldn't have any problem with compiling the code nor finding any headers.

BTW, have you got the patch for the `refsw` which contains an example of the package-config file + other needed changes? _(FYI, Broadcom accepted this patch and it should be available by default in the upcomming `refsw` release)_.

**Comment by bramoosterhuis** [comment]
Well it broke Nexus Playready 3 found in Nexus 18.1 which doen't have a package-config file. But instead of putting effort in a useless discussion I chose to fix it. 

Find the pull request here:
https://github.com/WebPlatformForEmbedded/OCDM-Playready-Nexus-SVP/pull/6

please check if it doesn't break your Playready 4 setup.

**Pull request merged**